src/mkMarkdown.ts

```diff
@@ -0,0 +1,25 @@
+import * as vscode from 'vscode';
+import { splitIfPanelExists } from './extension';
+
+export function mkMarkdown(): (...args: any[]) => any {
+  return async () => {
+    const currentPanel = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.viewColumn : undefined;
+    const uri = vscode.Uri.parse('markdown:' + 'L4'); // 'name of tab itself 
+    const doc = await vscode.workspace.openTextDocument(uri); // calls back into the provider
+
+    // ensure new panel opens instead of new tab
+    splitIfPanelExists(currentPanel);
+
+    await vscode.window.showTextDocument(doc, vscode.ViewColumn.Beside);
+  };
+}
+
+ // Register virtual doc provider
+ export const myScheme = 'markdown';
+ export const myProvider = new class implements vscode.TextDocumentContentProvider {
+
+   provideTextDocumentContent(uri: vscode.Uri): string {
+     let displayText = 'The Sale of Cabbages is Restricted. Everybody may sell the item, when the item is a cabbage and sale is onLegalDate or (unlikely) the seller has Exemption.';
```

Comment:
Can you please check if it's possible to move this string out to a separate file in a media folder?

Reply:
It works if I move the string into a file and then parse the file contents. But I wonder if that would defeat the purpose of using virtual documents, which take in take text and generate readonly documents (rather than read an existing file)? I'll push what I've done, but if we're going to display an existing static asset, it might be easier to open the file directly, rather than routing it through the virtualdocs api. It may be that I'm not understanding the virtualdocs api completely--I'm still not totally clear on how all the methods link up.